File: apps/mybookkeeper/frontend/src/shared/types/lease/extend-lease-request.ts
/**
 * Body for POST /signed-leases/{lease_id}/extend.
 *
 * Mirrors backend ``ExtendLeaseRequest``. ``new_ends_on`` is required and
 * must be strictly after the lease's current ``ends_on`` (the backend
 * enforces this and surfaces 409 ``NEW_END_DATE_NOT_AFTER_CURRENT``).
 * ``notes`` is optional free-text (max 2000 chars) that lands on the
 * rendered addendum. ``email_tenant`` triggers a best-effort tenant email
 * after the DB commit.
 */
export interface ExtendLeaseRequest {
  new_ends_on: string; // ISO-8601 YYYY-MM-DD
  notes?: string;
  email_tenant?: boolean;
}

File: apps/mybookkeeper/backend/app/schemas/leases/extend_lease_request.py
"""Request body for POST /signed-leases/{lease_id}/extend.

Extends a signed/active lease's end date. The service renders an extension
addendum PDF from the system-default boilerplate, attaches it to the lease,
appends a ``lease_term_versions`` row, and updates ``signed_leases.ends_on``.

``new_ends_on`` must be strictly after the current ``ends_on`` — the service
enforces this and the schema does not (the schema has no access to the
current value). ``notes`` is the optional free-text rationale that lands on
the rendered addendum. ``email_tenant`` triggers a best-effort send via the
existing ``lease_email_service`` after the DB commit.
"""
from __future__ import annotations

import datetime as _dt

from pydantic import BaseModel, ConfigDict, Field


class ExtendLeaseRequest(BaseModel):
    new_ends_on: _dt.date
    notes: str | None = Field(default=None, max_length=2000)
    email_tenant: bool = False

    model_config = ConfigDict(extra="forbid")
